docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +326,220 @@ recursion was explicitly requested.
+in which implementations are verified (default: N = 1).
+
+#### 22.2.2.8. `{:id <string>}`
+Assign a unique ID to an implementation to be used for verification
+result caching (default: "<impl. name>:0").
+
+#### 22.2.2.9. `{:timeLimit N}`
+Set the time limit for a given implementation.
+
+### 22.2.3. Verification attributes on functions
+
+#### 22.2.3.1. `{:builtin "spec"}`
+#### 22.2.3.2. `{:bvbuiltin "spec"}`
+Rewrite the function to built-in prover function symbol 'fn'.
+
+#### 22.2.3.3. `{:inline}`
+`{:inline}` or `{:inline true}` expands a function according to its definition before going to the prover.
+
+#### 22.2.3.4. `{:never_pattern true}`
+Terms starting with this function symbol will never be
+automatically selected as patterns. It does not prevent them
+from being used inside the triggers, and does not affect explicit
+trigger annotations. Internally it works by adding `{:nopats ...}`
+annotations to quantifiers.
+
+#### 22.2.3.5. `{:identity}`
+If the function annotated with `{:identity}` or `{:identity true}` has 1 argument and the use of it has type X->X for
+some X, then the abstract interpreter will treat the function as an
+identity function.  Note, the abstract interpreter trusts the
+attribute--it does not try to verify that the function really is an
+identity function on the abstract domain.
 
+### 22.2.4. Verification attributes on variables
+
+#### 22.2.4.1. `{:existential true}`
+Marks a global Boolean variable as existentially quantified. If
+used in combination with option `/contractInfer` Boogie will check
+whether there exists a Boolean assignment to the existentials
+that makes all verification conditions valid.  Without option
+`/contractInfer` the attribute is ignored.
+
+### 22.2.5. Verification attributes on assert statements {#sec-verification-attributes-on-assert-statements}
+
+To understand how to tweak verification,
+it is first useful to understand how Dafny verifies functions and methods.
+
+For every method, Dafny extracts _assertions_, as follows:
+
+* any explicit [`assert` statement](#sec-assert-statement) is _an assertion_.
+* A consecutive pair of lines in a [`calc` statement](#sec-calc-statement) forms _an assertion of their equality_.
+* Every function or method call with a [`requires` clause](#sec-requires-clause) yields _one assertion per clause_
+  (special cases such as sequence indexing come with a special require clause that the index is within bounds).
+* Assignments `o.f := E;` yield an _assertion_ that `o.f` is allowed by the enclosing [`modifies` clause](#sec-loop-framing).
+* Assignments by predicate `x :| P(x)` yield an _assertion_ that `exists x :: P(x)`.
+* Every [`ensures` clause](#sec-ensures-clause) yields an _assertion_ at the end of the method.
+* ...
+
+It is useful to mentally visualize all these assertions as a list (an _assertion batch_) that roughly follows the order in the code[^complexity-path-encoding],
```

Comment:
I strongly feel this kind of detailed content should be in the reference manual properly instead, rather than something dumped to the console for a user to read. I'm going to pause reviewing the rest of this to figure that out first.